packages/functional-tests/tests/settings/setup2faWithBackupCodes.spec.ts

```diff
@@ -0,0 +1,224 @@
+        testAccountTracker
+      );
+
+      await settings.goto();
+      await addTotpWithQrCodeAndBackupCodeChoice(settings, totp);
+
+      await page.waitForURL(/settings/);
+
+      await expect(settings.settingsHeading).toBeVisible();
+      await expect(settings.alertBar).toHaveText(
+        'Two-step authentication has been enabled'
+      );
+      await expect(settings.totp.status).toHaveText('Enabled');
+
+      await settings.disconnectTotp(); // Required before teardown
+
+      // No 2FA prompt on signin
+      await settings.signOut();
+      await signin.fillOutEmailFirstForm(credentials.email);
+      await signin.fillOutPasswordForm(credentials.password);
+      await expect(settings.settingsHeading).toBeVisible();
+    });
+
+    test('with manual key', async ({
+      target,
+      pages: { page, settings, totp, signin, configPage },
+      testAccountTracker,
+    }) => {
+      const config = await configPage.getConfig();
+      test.skip(
+        !config.featureFlags.updated2faSetupFlow,
+        'TODO in FXA-11941 - remove skip condition'
+      );
+      await signInAccount(target, page, settings, signin, testAccountTracker);
+
+      await settings.goto();
+      await addTotpWithManualCodeAndBackupCodeChoice(settings, totp);
+
+      await page.waitForURL(/settings/);
+
+      await expect(settings.settingsHeading).toBeVisible();
+      await expect(settings.alertBar).toHaveText(
+        'Two-step authentication has been enabled'
+      );
+      await expect(settings.totp.status).toHaveText('Enabled');
+
+      await settings.disconnectTotp(); // Required before teardown
+    });
+
+    test('delete account with 2FA enabled', async ({
+      target,
+      pages: { page, deleteAccount, settings, signin, totp, configPage },
+      testAccountTracker,
+    }) => {
+      const config = await configPage.getConfig();
+      test.skip(
+        !config.featureFlags.updated2faSetupFlow,
+        'TODO in FXA-11941 - remove skip condition'
+      );
+      const credentials = await signInAccount(
+        target,
+        page,
+        settings,
+        signin,
+        testAccountTracker
+      );
+
+      await settings.goto();
+      await addTotpWithQrCodeAndBackupCodeChoice(settings, totp);
+
+      await expect(settings.settingsHeading).toBeVisible();
+      await expect(settings.alertBar).toHaveText(
+        'Two-step authentication has been enabled'
+      );
+      await expect(settings.totp.status).toHaveText('Enabled');
+
+      await settings.deleteAccountButton.click();
+      await deleteAccount.deleteAccount(credentials.password);
+
+      await expect(
+        page.getByText('Account deleted successfully')
+      ).toBeVisible();
+    });
+  });
+});
+
+async function signInAccount(
+  target: BaseTarget,
+  page: Page,
+  settings: SettingsPage,
+  signin: SigninPage,
+  testAccountTracker: TestAccountTracker
+): Promise<Credentials> {
+  const credentials = await testAccountTracker.signUp();
```

Comment:
nit: I've noticed this in a bunch of tests, but I think it's weird to pass in the testAccountTracker... In the event we want to pick a different sign up option, this obscures what we want. e.g. What if we wanted to call `testAccountTracker.signUpSync()`?

IMO it's better if we just pass credentials into this function, so it doesn't care (ie it's not tightly coupled with `testAccountTracker.signUp()`).